### PR TITLE
[COLLECTIONS-761] - IteratorUtils.toList() will return empty list when size of iterator is 1

### PR DESCRIPTION
**Pull request by arturobernalg**
Add test with javadoc pointer to Jira and prove that it not reproducible

### CODE REVIEW

File: src/test/java/org/apache/commons/collections4/IteratorUtilsTest.java [hunk 1]
```diff
@@ -46,7 +47,9 @@
 import java.util.LinkedHashSet;
 
 import org.apache.commons.collections4.iterators.*;
+import org.apache.commons.collections4.list.GrowthList;
 import org.apache.commons.collections4.map.EntrySetToMapIteratorAdapter;
+import org.junit.Assert;
```
Comment (kinow) [inline src/test/java/org/apache/commons/collections4/IteratorUtilsTest.java:52]: Is this import being used?

File: src/test/java/org/apache/commons/collections4/IteratorUtilsTest.java [hunk 2]
```diff
@@ -46,7 +47,9 @@
 import java.util.LinkedHashSet;
 
 import org.apache.commons.collections4.iterators.*;
+import org.apache.commons.collections4.list.GrowthList;
```
Comment (kinow) [inline src/test/java/org/apache/commons/collections4/IteratorUtilsTest.java:50]: Is this import being used?

**Review by kinow — COMMENTED** [review]
The test code is simple, but at the moment it's not failing. From what I understood from the JIRA linked, it was supposed to test something that is not working? Unless we are adding the test for coverage?

File: src/test/java/org/apache/commons/collections4/IteratorUtilsTest.java (hunk 1 above)
Comment (arturobernalg) [reply src/test/java/org/apache/commons/collections4/IteratorUtilsTest.java]: True. Fix It

File: src/test/java/org/apache/commons/collections4/IteratorUtilsTest.java (hunk 2 above)
Comment (arturobernalg) [reply src/test/java/org/apache/commons/collections4/IteratorUtilsTest.java]: True. Fix It

**Comment by arturobernalg** [comment]
> The test code is simple, but at the moment it's not failing. From what I understood from the JIRA linked, it was supposed to test something that is not working? Unless we are adding the test for coverage?

Hi @kinow 

First that all Happy New Years. I tried to prove the Jira and was not possible. I think it's working correctly. For some reason @garydgregory  suggested adding the test. I guess its related with my last question about the regression

**Comment by kinow** [comment]
Hi @arturobernalg , happy 2021!

> I tried to prove the Jira and was not possible. I think it's working correctly.

From the JIRA description, it sounded to me like we would get an empty list when it was not desired. So that was the issue/problem we would need to fix for the next release.

> For some reason @garydgregory suggested adding the test. I guess its related with my last question about the regression

I think what Gary meant, is that along what I explained above, that if we indeed have an issue, it is really good if the person that created the JIRA issue can provide a unit test, and/or some small code snippet to show the developers what needs to be fixed.

For example, if you have a JIRA-1234 issue, where it says "Serializing an empty list results in an Exception", then it would be nice if the person reporting could include some code or unit test like

```java
@Test
public void testEmptyListSerializationIssue() {
    // NOTE: we are not testing assertThrows(() -> {...}) here, as the test is for "how should the API behave", and
    //             not "how the API is behaving now"
    UtilCode.serialize(new SomeCollectionsClass();
}
```

Hope it helps
Bruno

**Comment by arturobernalg** [comment]
Hi @kinow 

thank you very much. I understand now. 
I close the PR  and we wait for the person who reported the Jira send a test

**Pull request closed without merging**